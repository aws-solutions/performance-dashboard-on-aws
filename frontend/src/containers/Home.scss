.usa-search__submit-text {
    vertical-align: text-top !important;
}
