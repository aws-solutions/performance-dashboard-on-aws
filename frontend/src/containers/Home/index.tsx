import React, { useState, useEffect } from 'react';
import BadgerService from '../../services/badger-service';
import HomepageLayout from '../../layouts/Homepage';
import DashboardList from '../../components/DashboardList';

function Home() {
  const [dashboards, setDashboards] = useState([]);
  useEffect(() => {
    fetchDashboards();
  }, []);

  async function fetchDashboards() {
    const data = await BadgerService.fetchDashboards();
    setDashboards(data);
  }

  return (
    <HomepageLayout
      title="Performance Dashboard"
      subtitle="Find out how government services are performing and how we define our priorities to serve our citizens."
    >
      <DashboardList dashboards={dashboards} />
    </HomepageLayout>
  );
}

export default Home;
