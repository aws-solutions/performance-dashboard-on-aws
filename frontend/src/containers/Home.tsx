/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import { useHistory } from "react-router-dom";
import { usePublicHomepage } from "../hooks";
import { useTranslation } from "react-i18next";
import UtilsService from "../services/UtilsService";
import Accordion from "../components/Accordion";
import Search from "../components/Search";
import { PublicDashboard, LocationState, PublicTopicArea, Dashboard } from "../models";
import Spinner from "../components/Spinner";
import MarkdownRender from "../components/MarkdownRender";
import CardGroup from "../components/CardGroup";
import dayjs from "dayjs";
import LocalizedFormat from "dayjs/plugin/localizedFormat";
import "./Home.scss";

dayjs.extend(LocalizedFormat);

function Home() {
    const { homepage, loading } = usePublicHomepage();
    const { t } = useTranslation();
    const history = useHistory<LocationState>();
    const { Card, CardBody } = CardGroup;

    const onSearch = (query: string) => {
        history.push("/public/search?q=" + query);
    };

    const onClear = () => {
        history.push("/");
    };

    const topicareas = UtilsService.groupByTopicArea(homepage.dashboards);

    // Sort topic areas and their dashboards in ascending alphabetical order.
    topicareas.sort((a: PublicTopicArea, b: PublicTopicArea) => a.name.localeCompare(b.name));
    for (let topicArea of topicareas) {
        topicArea.dashboards?.sort((a: PublicDashboard, b: PublicDashboard) =>
            a.name.localeCompare(b.name),
        );
    }

    const getDashboardLink = (dashboard: PublicDashboard | Dashboard): string => {
        const link = dashboard.friendlyURL ? dashboard.friendlyURL : dashboard.id;
        return link;
    };

    return loading ? (
        <Spinner
            style={{
                position: "fixed",
                top: "30%",
                left: "50%",
            }}
            label={t("LoadingSpinnerLabel")}
        />
    ) : (
        <div className="usa-prose">
            <div className="grid-row">
                <div className="grid-col-12 tablet:grid-col-8">
                    <h1 className="font-sans-3xl line-height-sans-2 margin-top-2">
                        {homepage.title}
                    </h1>
                    <MarkdownRender
                        className="font-sans-lg usa-prose"
                        source={homepage.description}
                    />
                </div>
            </div>
            <div className="grid-row">
                <div className="grid-col-12 padding-y-3 usa-prose">
                    <Search
                        id="search"
                        onSubmit={onSearch}
                        size="big"
                        onClear={onClear}
                        query=""
                        results={homepage.dashboards.length}
                        wide
                    />
                </div>
            </div>
            <div className="grid-row">
                <div className="grid-col-12 usa-prose">
                    <Accordion>
                        {topicareas.map((topicarea) => (
                            <Accordion.Item
                                id={topicarea.id}
                                key={topicarea.id}
                                title={`${topicarea.name} (${topicarea.dashboards?.length ?? 0})`}
                                hidden={true}
                            >
                                <ul className="usa-card-group">
                                    {topicarea?.dashboards?.map((dashboard) => (
                                        <Card
                                            id={dashboard.id}
                                            title={dashboard.name}
                                            col={4}
                                            link={getDashboardLink(dashboard)}
                                        >
                                            <CardBody>
                                                <p className="text-base text-italic margin-bottom-2">
                                                    {`${t("LastUpdatedLabel")} ${dayjs(
                                                        dashboard.updatedAt,
                                                    ).fromNow()}`}
                                                </p>
                                                {dashboard.description && (
                                                    <p>
                                                        <MarkdownRender
                                                            source={dashboard.description}
                                                            className="font-sans-md usa-prose margin-top-0"
                                                        />
                                                    </p>
                                                )}
                                            </CardBody>
                                        </Card>
                                    ))}
                                </ul>
                            </Accordion.Item>
                        ))}
                    </Accordion>
                </div>
            </div>
        </div>
    );
}

export default Home;
