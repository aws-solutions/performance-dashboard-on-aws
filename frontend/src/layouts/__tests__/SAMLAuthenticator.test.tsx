/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import { render } from "@testing-library/react";
import { Route, Router, MemoryRouter } from "react-router-dom";
import { createMemoryHistory } from "history";
import withSAMLAuthenticator from "../SAMLAuthenticator";
import withPublicLayout from "../Public";
import Home from "../../containers/Home";

const history = createMemoryHistory();

describe("SAMLAuthenticatorLayout", () => {
    test("renders the component", async () => {
        const wrapper = render(
            <Router history={history}>
                <Route component={withSAMLAuthenticator(withPublicLayout(Home))} />
            </Router>,
            { wrapper: MemoryRouter },
        );
        expect(wrapper.container).toMatchSnapshot();
    });
});
