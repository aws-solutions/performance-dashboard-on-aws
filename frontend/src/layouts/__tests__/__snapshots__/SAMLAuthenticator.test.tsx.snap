// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SAMLAuthenticatorLayout renders the component 1`] = `
<div>
  <amplify-container>
    <amplify-authenticator>
      <amplify-sign-in
        slot="sign-in"
      />
    </amplify-authenticator>
  </amplify-container>
</div>
`;
